catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/util/EndpointUtil.java

```diff
@@ -539,4 +586,38 @@ public FilterBuilder getFilterBuilder() {
         return filterBuilder;
     }
 
+    /**
+     * Copy the attributes from a metacard to another metacard.
+     *
+     * @param sourceMetacard      the source metacard
+     * @param metacardType        copy all attributes represented by this metacard type
+     * @param destinationMetacard the destination metacard
+     */
+    public void copyAttributes(Metacard sourceMetacard, MetacardType metacardType,
+            Metacard destinationMetacard) {
+        metacardType.getAttributeDescriptors()
+                .forEach(attributeDescriptor -> Optional.ofNullable(sourceMetacard.getAttribute(
+                        attributeDescriptor.getName()))
+                        .ifPresent(attribute -> {
+                            List<Serializable> values = new LinkedList<>(attribute.getValues());
```

Comment:
❓ Is it safe to assume that every attribute will be multivalued? Or do we need to key off the attributeDescriptors.isMultivalued ?

Reply:
❓ Also why a linked list in this particular case?

Reply:
One solution to this problem could be:
```java
    public void copyAttributes(Metacard sourceMetacard, MetacardType metacardType,
            Metacard destinationMetacard) {
        metacardType.getAttributeDescriptors()
                .stream()
                .filter(name -> sourceMetacard.getAttribute(name) != null)
                .map(ad -> copyAttribute(ad, sourceMetacard))
                .forEach(destinationMetacard::setAttribute);
    }

    private Attribute copyAttribute(AttributeDescriptor attributeDescriptor, Metacard metacard) {
        String name = attributeDescriptor.getName();
        if (attributeDescriptor.isMultiValued()) {
            return new AttributeImpl(name, new ArrayList<>(metacard.getAttribute(name)));
        }
        return metacard.getAttribute(name);

    }

```

Reply:
Is it safe to assume isMultiValued is always set correctly? What if the field should be single valued but the source metacard has multiple values anyways? Since in this use case, I'm copying security fields, I'd hate to drop values even though they don't agree with the attribute schema.

Reply:
Yes. Multivalued should always be correct. If it isn't there are a lot bigger problems at hand.

Reply:
ok, switched over to your solution